service/controller/resource/cleanuploadbalancers/delete.go

```diff
@@ -19,31 +20,64 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	lbState, err := r.clusterLoadBalancers(ctx, cr)
-	if err != nil {
-		return microerror.Mask(err)
-	}
+	// delete classic load-balancers
+	{
+		lbState, err := r.clusterClassicLoadBalancers(ctx, cr)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if lbState != nil && len(lbState.LoadBalancerNames) > 0 {
+			r.logger.Debugf(ctx, "deleting %d load balancers", len(lbState.LoadBalancerNames))
+
+			cc, err := controllercontext.FromContext(ctx)
```

Comment:
We can pull out the controller context at the top like in any other handler.

Reply:
fixed